src/domain/challenge/challenge/challenge.service.ts

```diff
@@ -75,6 +86,93 @@ export class ChallengeService {
     return challenge;
   }
 
+  async updateChallenge(
+    challengeData: UpdateChallengeInput
+  ): Promise<IChallenge> {
+    const challenge = await this.getChallengeOrFail(challengeData.ID);
+
+    const newName = challengeData.name;
+    if (newName) {
+      if (!(newName === challenge.name)) {
```

Comment:
Style-wise this is an odd check - might be more readable with with !== instead of !(... === ...)

Reply:
Will fix.